pkg/trustedresources/verify.go

```diff
@@ -40,11 +43,28 @@ const (
 
 // VerifyTask verifies the signature and public key against task.
 // source is from ConfigSource.URI, which will be used to match policy patterns. k8s is used to fetch secret from cluster
-func VerifyTask(ctx context.Context, taskObj v1beta1.TaskObject, k8s kubernetes.Interface, source string, policies []*v1alpha1.VerificationPolicy) error {
-	matchedPolicies, err := matchedPolicies(taskObj.TaskMetadata().Name, source, policies)
+func VerifyTask(ctx context.Context, taskObj v1beta1.TaskObject, k8s kubernetes.Interface, source string, verificationpolicies []*v1alpha1.VerificationPolicy) error {
+	matchedPolicies, err := getMatchedPolicies(taskObj.TaskMetadata().Name, source, verificationpolicies)
 	if err != nil {
-		return err
+		if errors.Is(err, ErrNoMatchedPolicies) {
+			// skip the verification if no policies are found and trusted-resources-verification-no-match-policy is set to ignore or warn
+			switch config.GetVerificationNoMatchPolicy(ctx) {
+			case config.IgnoreNoMatchPolicy:
+				return nil
+			case config.WarnNoMatchPolicy:
+				logger := logging.FromContext(ctx)
+				logger.Warnf("failed to get matched policies: %v", err)
+				return nil
+			case config.FailNoMatchPolicy:
+				fallthrough
+			default:
+				// if the config is not "ignore" or "warn", then it should be "fail" and return error
+				return fmt.Errorf("VerifyPipeline failed to get matched policies: %w", err)
+			}
+		}
+		return fmt.Errorf("failed to get matched policies: %w", err)
```

Comment:
Old comment got deleted by the force push :(

What I meant was that you can do something like:

```suggestion
		if errors.Is(err, ErrNoMatchedPolicies) {
			// skip the verification if no policies are found and trusted-resources-verification-no-match-policy is set to ignore or warn
			switch config.GetVerificationNoMatchPolicy(ctx) {
			case config.IgnoreNoMatchPolicy:
				return nil
			case config.WarnNoMatchPolicy:
				logger := logging.FromContext(ctx)
				logger.Warnf("failed to get matched policies: %v", err)
				return nil
			}
		}
		return fmt.Errorf("failed to get matched policies: %w", err)
```

instead of having 2 errors with slightly different messages.

Reply:
Oh I thought we must have a `default` in the `switch` and another reason is for readability? Because we support 3 values for the  `trusted-resources-verification-no-match-policy`

Reply:
No hard rule here. I think it depends on what error you want to return back. IIRC one of the problems before was that we could fail open if we had an unrecognized config state. The default lets you prevent that, but it also has a trade off of now you have 2 separate errors that could be returned and diverge in content (you even see it here - they have slightly different messages).

That said, maybe there's an argument to be made that we **should** return a different error depending on whether you consider returning 0 policies a failure or not. 🤷 

If you want to return separate errors, keep what you have and update the error message. If you want them to have the same error then have it converge on the same error return. Dealer's choice.

Reply:
Thanks! Adopted the suggestions